File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
  let answer = strings.sort((a, b) => {
     if(a[n] > b[n]) return 1;
      if(a[n] < b[n]) return -1;
      //내림차순으로 하기위에 뒤에값 b가 클때를 음수를 준다.
      if(a[n] === b[n]){
          if(a>b) return 1;
          if(a<b) return -1;
          //내림차순으로 하기위에 뒤에값 b가 클때를 음수를 준다.
          return 0;
      }
  });
  
  return answer;
}

File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;

class Solution {
 public String[] solution(String[] strings, int n) {
      List<String> list = Arrays.asList(strings);
      list.sort((a, b) -> {
          int result = (a.split(""))[n].compareTo((b.split(""))[n]);
          if(result == 0)
              return a.compareTo(b);
          return result;
      });
      return list.toArray(new String[0]);
  }
}